fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBQueriedRecord.java

```diff
@@ -278,7 +353,45 @@ public FDBStoredRecord<M> getStoredRecord() {
 
         @Nullable
         @Override
-        public Map<String, FDBStoredRecord<? extends Message>> getConstituents() {
+        public FDBSyntheticRecord getSyntheticRecord() {
+            return null;
+        }
+
+        @Nullable
+        @Override
+        public String getConstituentName() {
+            return null;
+        }
+
+        @Nullable
+        @Override
+        @SuppressWarnings("unchecked")
+        public FDBQueriedRecord<M> getConstituent(@Nonnull String constituentName) {
```

Comment:
Ah, so we have synthetic records that are covered records. That's why you expose `getConstituent()` directly. Correct?

Reply:
Correct. There are already `FDBQueriedRecord`s that aren't `FDBStoredRecord`, but implement some similar functionality when coming from covering scans. And there are already `FDBQueriedRecord`s for synthetic record types that aren't `FDBSyntheticRecord` when coming from covering scans. These changes add some additional constituent-related operations around these, staying within the `FDBQueriedRecord` realm.

Fundamentally, having to pretend that covering scans produce records at all is problematic. It gets somewhat worse with join indexes. But fixing that is a comprehensive change.